src/librbd/io/QosImageDispatch.cc

```diff
@@ -278,7 +283,8 @@ bool QosImageDispatch<I>::needs_throttle(
     Context* on_dispatched) {
   auto cct = m_image_ctx->cct;
   auto extent_length = get_extent_length(image_extents);
-  bool all_qos_flags_set = false;
+  bool all_qos_flags_set = true;
```

Comment:
I don't think you need to change this. You could initialize to false and only update at the end of the function if you call set_throttle_flags